test/sql/optimizer/expression/test_equal_or_null_optimization.test

```diff
@@ -0,0 +1,35 @@
+# name: test/sql/optimizer/expression/test_equal_or_null_optimization.test
+# description: Test a=b OR (a IS NULL AND b IS NULL) to a IS NOT DISTINCT FROM b
+# group: [expression]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+START TRANSACTION;
+
+statement ok
+CREATE TABLE test(i INTEGER, j INTEGER);
+
+statement ok
+INSERT INTO test VALUES (1,1), (2,2), (NULL,NULL)
+
+statement ok
+PRAGMA explain_output = 'OPTIMIZED_ONLY';
+
+query I rowsort
+SELECT i FROM test WHERE (i=j) OR (i IS NULL AND j IS NULL);
+----
+1
+2
+NULL
+
+query I rowsort
+SELECT i FROM test WHERE (i IS NULL AND j IS NULL) OR (i=j);
+----
+1
+2
+NULL
+
+statement ok
```

Comment:
Can we add some tests that cover the rejected cases?

Reply:
Just added them. I looked on the wrong test directory :)